### PR TITLE
fixing gcp inv plugin

### PR DESCRIPTION
**Pull request by bcoca**
##### ISSUE TYPE

- Bugfix Pull Request
##### COMPONENT NAME

inventory/gcp_compute

### CODE REVIEW

**Comment by AlanCoding** [comment]
testing

```
ansible-inventory -i gcp_compute.yml --list --export -vvv
```

```
 [WARNING]:  * Failed to parse /Users/alancoding/Documents/repos/ansible-inventory-file-examples/private/gce/gcp_compute.yml with auto plugin: local variable 'zones'
referenced before assignment

  File "/Users/alancoding/Documents/repos/ansible/lib/ansible/inventory/manager.py", line 272, in parse_source
    plugin.parse(self._inventory, self._loader, source, cache=cache)
  File "/Users/alancoding/Documents/repos/ansible/lib/ansible/plugins/inventory/auto.py", line 58, in parse
    plugin.parse(inventory, loader, path, cache=cache)
  File "/Users/alancoding/Documents/repos/ansible/lib/ansible/plugins/inventory/gcp_compute.py", line 387, in parse
    for zone in zones:
```

**Comment by AlanCoding** [comment]
This new error is non-trivial, so I did make sure to re-test with the `devel` branch. This is using an inventory file that does the serviceaccount authentication.

```
 [WARNING]:  * Failed to parse /Users/alancoding/Documents/repos/ansible-inventory-file-examples/private/gce/gcp_compute.yml with auto plugin: ('invalid_scope: h is not a valid
audience string.', u'{\n  "error": "invalid_scope",\n  "error_description": "h is not a valid audience string."\n}')

  File "/Users/alancoding/Documents/repos/ansible/lib/ansible/inventory/manager.py", line 272, in parse_source
    plugin.parse(self._inventory, self._loader, source, cache=cache)
  File "/Users/alancoding/Documents/repos/ansible/lib/ansible/plugins/inventory/auto.py", line 58, in parse
    plugin.parse(inventory, loader, path, cache=cache)
  File "/Users/alancoding/Documents/repos/ansible/lib/ansible/plugins/inventory/gcp_compute.py", line 392, in parse
    resp = self.fetch_list(params, link, query)
  File "/Users/alancoding/Documents/repos/ansible/lib/ansible/plugins/inventory/gcp_compute.py", line 170, in fetch_list
    response = auth.get(link, params={'filter': query})
  File "/Users/alancoding/Documents/repos/ansible/lib/ansible/module_utils/gcp_utils.py", line 86, in get
    return self.session().get(url, **kwargs)
  File "/Users/alancoding/.virtualenvs/ansible/lib/python2.7/site-packages/requests/sessions.py", line 546, in get
    return self.request('GET', url, **kwargs)
  File "/Users/alancoding/.virtualenvs/ansible/lib/python2.7/site-packages/google/auth/transport/requests.py", line 198, in request
    self._auth_request, method, url, request_headers)
  File "/Users/alancoding/.virtualenvs/ansible/lib/python2.7/site-packages/google/auth/credentials.py", line 122, in before_request
    self.refresh(request)
  File "/Users/alancoding/.virtualenvs/ansible/lib/python2.7/site-packages/google/oauth2/service_account.py", line 322, in refresh
    request, self._token_uri, assertion)
  File "/Users/alancoding/.virtualenvs/ansible/lib/python2.7/site-packages/google/oauth2/_client.py", line 145, in jwt_grant
    response_data = _token_endpoint_request(request, token_uri, body)
  File "/Users/alancoding/.virtualenvs/ansible/lib/python2.7/site-packages/google/oauth2/_client.py", line 111, in _token_endpoint_request
    _handle_error_response(response_body)
  File "/Users/alancoding/.virtualenvs/ansible/lib/python2.7/site-packages/google/oauth2/_client.py", line 61, in _handle_error_response
    error_details, response_body)
```

This works in the `devel` branch.

File: lib/ansible/plugins/inventory/gcp_compute.py [hunk 1]
```diff
@@ -338,27 +347,18 @@ def parse(self, inventory, loader, path, cache=True):
...
-            config_data['filters'] = None
-
-        projects = config_data['projects']
-        zones = config_data.get('zones')
-        config_data['scopes'] = ['https://www.googleapis.com/auth/compute']
+        # setup parameters as expected by 'fake module class' to reuse module_utils w/o changing the API
+        params = {
+            'filters': self.get_option('filters'),
+            'projects': self.get_option('projects'),
+            'scopes': 'https://www.googleapis.com/auth/compute',
```
Comment (AlanCoding) [inline lib/ansible/plugins/inventory/gcp_compute.py]: this changed from a list to a string.

Perhaps this is related to the "h" message in the error.

Comment (bcoca) [reply lib/ansible/plugins/inventory/gcp_compute.py]: good catch, that was driving me nuts

also noticed, this was hinted at being 'configurable' but never was

**Review by AlanCoding — APPROVED** [review]
I had a hard time verifying this, because I got a lot of read timeouts... but I think this should be a transient gce issue, and this eventually worked.

**Review by maxamillion — APPROVED** [review]

**Pull request merged**